lessons/git/intro/slides-2017.Rmd

```diff
@@ -0,0 +1,305 @@
+```bash
+cd ~/Documents
+# for windows users with full path
+cd C:/Users/user_name/Documents
+# create a new folder 
+# use the same name as your remote repo on GitHub
+mkdir git-training
+# move to the new folder
+cd git-training
+```
+## Convert folder to **git** repo
+```bash
+git init
+```
+## check your folder to find git configuration files/folders
+This folder is now a git repo (short for repository)
+
+# Using GitHub with Git
+If you have GitHub, you can add the repository you cloned/created to it and visualize the changes
+
+Remember, you installed this earlier?
+
+# Configuring your repo for use with GitHub on the cloud
+In order to use this local repo with the one you created earlier we need to set the remote configuration
+```bash
+git remote -v
+```
+will tell you what is the remote corresponding to your local repo
+
+If it is empty, you can setup a remote using 
+```bash
+git remote add remote_name remote_url
+```
+
+Before you do this, you need to have an account on GitHub and a repo on it
+
+# Configuring your repo for use with GitHub on the cloud
+* Copy the repo url from the repo creation page and use it in
+```bash
+git remote add origin copy_url_here
+```
+
+* Check your remote configuration again
+```bash
+git remote -v
+```
+
+# Adding new files to a repo
+* create a text file with
+    + your name
+    + program
+    + research area
+* save the file as `my_cred` in the git repo you just created
+
+# Visualize changes to files
+You can see the changes you made to this file in GitHub
+\begin{block}
+{New files} on GitHub can be seen with a \textcolor{green}{green '+'} to their right 
+\end{block}
+
+This indicates the file is new and is `untracked` by git
+
+\begin{block}
+{Old modified files} can be seen with a \textcolor{gray}{gray circle} to their right
+\end{block}
+
+This indicates the file is already `tracked` and there are `uncommited` changes
+```bash
+git status
+``` 
+can give information on `tracked`, `untracked` and `indexed` files
+
+# Adding Files to git index
+* add the file to the git index
+```bash
+git add ~/Documents/new_git_repo/my_cred
+# you can also use wildcards instead of full name
+git add *my_cred
+```
+* check file status again 
+```bash
+git status
+```
+Notice the file has changed from \textcolor{red}{red} to \textcolor{green}{green}
+
+* let us `commit` this change to repo history 
+```bash
+git commit -m "initial commit with my info"
+```
+* Now your file is `tracked` by git
+
+# `push` to your new repo on GitHub
+You can move all your **committed** changes to the remote repo
+```bash
+git push origin
+```
+You can also get all **commits** from the remote repo
+```bash
+git pull origin
+```
+Make sure your `remote` is set before you do this
+
+# Cloning an Existing `git` repo
+We can copy or clone existing repositories from GitHub using
+```bash
+git clone
+```
+Let us do this for the UofT Coders repo available at
+[UofT Coders git repo](https://github.com/UofTCoders/studyGroup)
```

Comment:
Again, let's use a sandbox repo.

Reply:
This would require 2 sandbox repos though (One for local with remote configuration testing and other for cloning from remote). I guess that is ok.